### PR TITLE
Issue #2: automatically select best model for target variable

### PR DESCRIPTION
**Pull request by rachelt44**

### CODE REVIEW

File: src/perform_predict/predict.py [hunk 1]
```diff
@@ -97,23 +97,38 @@ def _create_input_space(input_config: Dict[str, list[Dict[str, Any]]],
...
 
 
+def _get_highest_ranked_estimator(estimator_path: str, target_variable: str) -> str:
+    ranking_file = os.path.join(estimator_path, constants.AM_RANKINGS_FILE)
+    if not os.path.exists(ranking_file):
+        logger.error("Failed to locate model ranking file")
+        raise Exception
+    ranking_df = pd.read_csv(ranking_file)
+    ranking_df_var = ranking_df.loc[ranking_df[constants.AM_COL_TARGET] == target_variable]
+    ranking_df_var['combined_rank'] = ranking_df_var[constants.AM_COL_RANK_MAPE] + \
```
Comment (drbwa) [inline src/perform_predict/predict.py:119]: I am not sure about having a combined rank across two measures of prediction
error together with R^2 which speaks more to correlation. Best combined rank of
MAPE and NRMSE seems less problematic to me. Maybe we should consider not
including R^2 in this decision. What do you think?

**Review by drbwa — CHANGES_REQUESTED** [review]

**Review by drbwa — CHANGES_REQUESTED** [review]
Much-needed feature. Looks good in general. Just have one concern about the approach to come up with a `combined_ranking` across MAPE, NRMSE, and R^2. Let me know what you think.

**Comment by rachelt44** [comment]
> Much-needed feature. Looks good in general. Just have one concern about the approach to come up with a `combined_ranking` across MAPE, NRMSE, and R^2. Let me know what you think.

@drbwa I agree, removed R^2 from the combined ranking

File: src/perform_predict/predict.py (hunk 1 above)
Comment (rachelt44) [reply src/perform_predict/predict.py:119]: Good point. Removed R^2 from combined ranking.

**Pull request merged**